scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2999,11 +2999,13 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     val paramss = paramClauses(ownerIsType = false).require[List[List[Term.Param]]]
 
     def onlyLastParameterCanBeRepeated(params: List[Term.Param]): Unit = {
-      params.reverse.tail.foreach(p =>
-        if (p.decltpe.exists(_.isInstanceOf[Type.Repeated])) {
-          syntaxError("*-parameter must come last", p)
-        }
-      )
+      params
+        .take(params.length - 1)
```

Comment:
This will also allocate a new list, but you can avoid that by making it an iterator first

Reply:
good catch with this allocation, I will try use `iterator`

Reply:
changed to iterator :+1: